src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -66,8 +66,7 @@ public List<MetricFamilySamples> collect() {
                 continue;
             }
             LOG.info("Kafka metric {} is allowed", name);
-            LOG.info("labels " + metricName.tags());
-            MetricFamilySamples sample = convert(name, metricName.description(), kafkaMetric, metricName.tags());
+            MetricFamilySamples sample = convert(name, metricName.description(), kafkaMetric, metricName.tags(), metricName);
```

Comment:
this method's api look weird. passing metricName and metricName.description() and metricName.tags()

Reply:
@k-wall Does this satisfy the code smell? The only thing I'm not sure about is using `metric.metricName().description()` instead of `help`

   ```
               MetricFamilySamples sample = convert(prometheusMetricName,  kafkaMetric, metricName);

   private static MetricFamilySamples convert(String name, KafkaMetric metric, MetricName metricName) {
        Map<String, String> sanitizedLabels = metricName.tags().entrySet().stream()
                .collect(Collectors.toMap(
                        e -> Collector.sanitizeMetricName(e.getKey()),
                        Map.Entry::getValue,
                        (v1, v2) -> {
                            LOG.warn("Ignoring metric value duplicate key {}", v1);
                            return v1;
                        },
                        LinkedHashMap::new));
        Object valueObj = metric.metricValue();
        double value;
        if (valueObj instanceof Number) {
            value = ((Number) valueObj).doubleValue();
        } else {
            value = 1.0;
            String attributeName = metricName.name();
            sanitizedLabels.put(attributeName, String.valueOf(valueObj));
        }

        return new MetricFamilySamplesBuilder(Type.GAUGE, metric.metricName().description())
                .addSample(name, value, sanitizedLabels)
                .build();
    }

Reply:
> Does this satisfy the code smell? 

Yes, but rename the formal parameter `name` in the method declaration too.

> The only thing I'm not sure about is using metric.metricName().description() as help. In YammerMetricsCollector, the help argument is an empty String.

My comment was against KafkaMetricsCollector only, not YammerMetricsCollector.